src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4477,6 +4478,21 @@ void os::Linux::numa_init() {
   }
 }
 
+// Check numa dependent syscalls
+bool os::Linux::numa_syscall_check() {
+  // NUMA APIs depend on several syscalls. E.g., get_mempolicy is required for numa_get_membind and
+  // numa_get_interleave_mask. But these dependent syscalls can be unsupported for various reasons.
+  // Especially in dockers, get_mempolicy is not allowed with the default configuration. So it's necessary
+  // to check whether the syscalls are available. Currently, only get_mempolicy is checked since checking
+  // others like mbind would cause unexpected side effects.
+  int dummy = 0;
+  if (syscall(SYS_get_mempolicy, &dummy, NULL, 0, (void*)&dummy, 3) == -1) {
```

Comment:
Is this SYS_get_mempolicy symbol guaranteed to be available on our supported Linux versions? See earlier in the file for how we handle SYS_gettid and SYS_getcpu.

Reply:
> Is this SYS_get_mempolicy symbol guaranteed to be available on our supported Linux versions? See earlier in the file for how we handle SYS_gettid and SYS_getcpu.

I think it's OK to use the symbol directly since we have used it in ZGC [1].
Thanks.

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/os/linux/gc/z/zSyscall_linux.cpp#L39